Fix overflowing dialog content caused by long email address

dcalhoun opened the pull request:
### Fix
Fixes #2709. Allow the dialog to wrap the email address to prevent the content
from overflowing the dialog container.

| Before | After |
| --- | --- |
| <img width="1226" alt="109825032-07a40a00-7bff-11eb-8895-837555f564af" src="https://user-images.githubusercontent.com/438664/109834478-da0f8e80-7c07-11eb-86da-efa03c102c53.png"> | <img width="1552" alt="Screen Shot 2021-03-03 at 09 59 03" src="https://user-images.githubusercontent.com/438664/109834509-e1cf3300-7c07-11eb-8469-c53ae79e72ba.png"> |

### Test
1. Visit the Simplenote desktop or web app.
1. Sign up with a fairly long email address.

Expected outcome: The email address wraps within the dialog rather than
overflowing the dialog.

### Release

`RELEASE-NOTES.md` was updated with:
- Fix overflowing dialog content when long email address is used during sign up.

dcalhoun committed:
Fix overflowing dialog content caused by long email address

Allow the dialog to wrap the email address to prevent the content from
overflowing the dialog container.

codebykat reviewed (changes requested):
Unfortunately, it does this in Firefox (or presumably anywhere the window/zoom size doesn't work out right). I think we need to add a span around the email address and just put `break-all` on that.

<img width="380" alt="Screen Shot 2021-03-03 at 10 07 35 PM" src="https://user-images.githubusercontent.com/52152/109919110-fc87c300-7c6c-11eb-9e0d-5622bd270323.png">

codebykat commented:
Somewhat absurdly, putting the style on the `<strong>` even works. I think we should make the break before the email an explicit `<br>` as well, that will help matters.

<img width="626" alt="Screen Shot 2021-03-03 at 10 10 16 PM" src="https://user-images.githubusercontent.com/52152/109919338-59837900-7c6d-11eb-8901-444bd94313d1.png">

<img width="379" alt="Screen Shot 2021-03-03 at 10 10 20 PM" src="https://user-images.githubusercontent.com/52152/109919345-5c7e6980-7c6d-11eb-8c3c-771238a7a04c.png">

dcalhoun committed:
Place email on separate line to reduce likelihood of wrapping

Long email addresses must wrap to avoid overflowing the modal width.
Placing the email on a new line provides more space for the email
address.

dcalhoun commented:
@codebykat good catch. Just pushed a change. Is the image below what you were envisioning? New line for the email to avoid likelihood of wrapping.

<img width="1136" alt="Screen Shot 2021-03-04 at 08 33 28" src="https://user-images.githubusercontent.com/438664/109979581-a8a8c880-7cc4-11eb-8eba-86be09578945.png">

codebykat reviewed (approved):
Looks great now 👍 

<img width="385" alt="Screen Shot 2021-03-04 at 9 25 05 AM" src="https://user-images.githubusercontent.com/52152/110004008-d2b1b900-7ccb-11eb-8fc1-c99579350b50.png">
